### PR TITLE
cmd/prometheus: Issue a warning on 32 bit archs

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: cmd/prometheus/main.go [hunk 1]
```diff
@@ -344,6 +345,10 @@ func main() {
 	klog.SetLogger(log.With(logger, "component", "k8s_client_runtime"))
 
 	level.Info(logger).Log("msg", "Starting Prometheus", "version", version.Info())
+	if bits.UintSize < 64 {
+		level.Warn(logger).Log("msg", "This Prometheus binary hasn't been compiled for a 64-bit architecture. Due to memory mapping constraints of 32-bit systems, it is highly recommended to switch to a 64-bit build of Prometheus.", "GOARCH", runtime.GOARCH)
```
Comment (brian-brazil) [inline cmd/prometheus/main.go:349]: It's a virtual memory constraint rather than an mmap constraint. mmap itself works fine on 32bit.

**Comment by brian-brazil** [comment]
:+1:

**Comment by roidelapluie** [comment]
@brian-brazil can you +1 the 2 las minor changes I made?

- used `binary` twice in the sentence instead of `binary` & `build`
- use `has not` instead of `hasn't`

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**